world-chain-builder/crates/world/node/tests/e2e.rs

```diff
@@ -149,94 +131,47 @@ impl WorldChainBuilderTestContext {
         } = builder.launch().await?;
 
         let test_ctx = NodeTestContext::new(node, optimism_payload_attributes).await?;
-
         Ok(Self {
-            pbh_wallets: wallets,
-            tree,
+            signers: (0..5),
             tasks,
             node: test_ctx,
-            identities,
         })
     }
 
-    pub async fn raw_pbh_tx_bytes(
-        &self,
-        signer: PrivateKeySigner,
-        pbh_nonce: u8,
-        tx_nonce: u64,
-    ) -> Bytes {
-        let tx = tx(DEV_CHAIN_ID, None, tx_nonce);
-        let envelope = TransactionTestContext::sign_tx(signer.clone(), tx).await;
-        let raw_tx = envelope.encoded_2718();
-        let mut data = raw_tx.as_ref();
-        let recovered = PooledTransactionsElement::decode_2718(&mut data).unwrap();
-        let pbh_payload = self.valid_proof(
-            signer.address(),
-            recovered.hash().as_slice(),
-            chrono::Utc::now(),
-            pbh_nonce,
-        );
-
-        let world_chain_pooled_tx_element = WorldChainPooledTransactionsElement {
-            inner: recovered,
-            pbh_payload: Some(pbh_payload.clone()),
-        };
-
-        let mut buff = Vec::<u8>::new();
-        world_chain_pooled_tx_element.encode_2718(&mut buff);
-        buff.into()
-    }
-
-    fn valid_proof(
-        &self,
-        identity: Address,
-        tx_hash: &[u8],
-        time: chrono::DateTime<Utc>,
-        pbh_nonce: u8,
-    ) -> PbhPayload {
-        let external_nullifier =
-            ExternalNullifier::with_date_marker(DateMarker::from(time), pbh_nonce).to_string();
-
-        self.create_proof(identity, external_nullifier, tx_hash)
-    }
-
-    fn create_proof(
-        &self,
-        mut identity: Address,
-        external_nullifier: String,
-        signal: &[u8],
-    ) -> PbhPayload {
-        let external_nullifier: ExternalNullifier = external_nullifier.parse().unwrap();
-
-        let idx = self.identities.get(&identity).unwrap();
-        let secret = identity.as_mut_slice();
-        // generate identity
-        let id = Identity::from_secret(secret, None);
-        let merkle_proof = self.tree.proof(*idx);
-
-        let signal_hash = hash_to_field(signal);
-        let external_nullifier_hash = external_nullifier.to_word();
-        let nullifier_hash = generate_nullifier_hash(&id, external_nullifier_hash);
-
-        let proof = Proof(
-            generate_proof(&id, &merkle_proof, external_nullifier_hash, signal_hash).unwrap(),
+    pub async fn raw_pbh_tx_bytes(&self, acc: u32, pbh_nonce: u8, tx_nonce: u64) -> Bytes {
+        let dt = chrono::Utc::now();
+        let dt = dt.naive_local();
+
+        let month = dt.month() as u8;
+        let year = dt.year() as u16;
+
+        let ext_nullifier = ExternalNullifier::v1(month, year, pbh_nonce);
+        let (uo, proof) = user_op()
+            .acc(acc)
+            .external_nullifier(ext_nullifier)
+            .nonce(U256::from(pbh_nonce))
```

Comment:
maybe another param? User op nonce? Shouldn't affect the tests but just to avoid confusion